src/visualizer/raysAppFunctions/updateScenario.m

```diff
@@ -113,68 +119,170 @@ function setupTimestepInfo(app)
 
 function extractQdFilesInfo(app)
 qdFiles = dir(sprintf('%s/QdFiles',app.ns3Path));
-
-for i = 1:length(qdFiles)
-    token = regexp(qdFiles(i).name,'Tx(\d+)Rx(\d+).txt','tokens');
-    if isempty(token)
-        continue
-    end
-    
-    % else
-    Tx = str2double(token{1}{1}) + 1;
-    Rx = str2double(token{1}{2}) + 1;
-    
-    qd = readQdFile(sprintf('%s/%s',...
-        qdFiles(i).folder,qdFiles(i).name));
-    
-    for t = 1:length(qd)
-        app.timestepInfo(t).qdInfo(Tx,Rx) = qd(t);
-    end
+qdFileName = qdFiles(3).name;
+[~,~,extension]=fileparts(qdFileName);
+switch extension
+    case '.json'
+        qdOutput = readQdJsonFile(sprintf('%s/QdFiles/qdOutput.json',...
+            app.ns3Path));
+        for i = 1:size(qdOutput,2)
+            for timestep = 1:size(qdOutput(1).Delay,1)
+                out = struct('TX', cell(1,1), ...
+                    'RX', cell(1,1), ...
+                    'PAA_TX', cell(1,1), ...
+                    'PAA_RX', cell(1,1), ...
+                    'Delay', cell(1,1), ...
+                    'Gain', cell(1,1), ...
+                    'Phase', cell(1,1), ...
+                    'AODEL', cell(1,1), ...
+                    'AODAZ', cell(1,1), ...
+                    'AOAEL', cell(1,1), ...
+                    'AOAAZ', cell(1,1) ...
```

Comment:
there's no reason to put struct's fields in capital letters, i'd rather use camelCase

Reply:
Check email. Please leave it like this.